integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/sdx/SdxResizeTests.java

```diff
@@ -35,34 +41,65 @@
+        AtomicReference<String> expectedName = new AtomicReference<>();
         testContext
                 .given(sdx, SdxInternalTestDto.class)
                 .when(sdxTestClient.createInternal(), key(sdx))
                 .await(SdxClusterStatusResponse.RUNNING, key(sdx))
                 .awaitForHealthyInstances()
                 .then((tc, testDto, client) -> {
                     expectedShape.set(sdxUtil.getShape(testDto, client));
+                    expectedCrn.set(sdxUtil.getCrn(testDto, client));
+                    expectedName.set(sdx);
                     return testDto;
                 })
                 .when(sdxTestClient.resize(), key(sdx))
                 .await(SdxClusterStatusResponse.STOP_IN_PROGRESS, key(sdx).withWaitForFlow(Boolean.FALSE))
-                .await(SdxClusterStatusResponse.DATALAKE_DETACHED)
-                .await(SdxClusterStatusResponse.RUNNING, key(resizedSdx))
+                .await(SdxClusterStatusResponse.STACK_CREATION_IN_PROGRESS)
+                .await(SdxClusterStatusResponse.RUNNING, key(sdx))
                 .awaitForHealthyInstances()
-                .then((tc, testDto, client) -> {
-                    actualShape.set(sdxUtil.getShape(testDto, client));
-                    return testDto;
-                })
-                .then((tc, testDto, client) ->  {
-                    if (actualShape.get().equals(expectedShape.get())) {
-                        throw new TestFailException("SDX shape did not change after re-sizing");
-                    }
-                    return testDto;
-                })
+                .then((tc, dto, client) -> validateStackCrn(expectedCrn, dto))
+                .then((tc, dto, client) -> validateCrn(expectedCrn, dto))
+                .then((tc, dto, client) -> validateShape(expectedShape, dto))
+                .then((tc, dto, client) -> validateClusterName(expectedName, dto))
                 .validate();
     }
+
+    private SdxInternalTestDto validateStackCrn(AtomicReference<String> originalCrn, SdxInternalTestDto dto) {
+        String newCrn = dto.getResponse().getStackV4Response().getCrn();
+        Log.log(LOGGER, format(" Stack new crn: %s ", newCrn));
+        if (!newCrn.equals(originalCrn.get())) {
+            throw new TestFailException(" The stack CRN has changed to: " + newCrn + " instead of: " + originalCrn.get());
+        }
+        return dto;
+    }
+
+    private SdxInternalTestDto validateCrn(AtomicReference<String> originalCrn, SdxInternalTestDto dto) {
+        String newCrn = dto.getResponse().getCrn();
+        Log.log(LOGGER, format(" New crn: %s ", newCrn));
+        if (!newCrn.equals(originalCrn.get())) {
+            throw new TestFailException(" The stack CRN has changed to: " + newCrn + " instead of: " + originalCrn.get());
+        }
+        return dto;
+    }
+
+    private SdxInternalTestDto validateShape(AtomicReference<String> originalCrn, SdxInternalTestDto dto) {
```

Comment:
Here the `expectedShape` or based on previous pattern `originalShape` should be. However I cannot see the later usage of this parameter.

Reply:
that code was stale. removed it.